Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -74,16 +87,50 @@ bool TcpClient::sendString(const String& data, bool forceCloseAfterSent /* = fal
 	return send(data.c_str(), data.length(), forceCloseAfterSent);
 }
 
+bool TcpClient::send(IDataSourceStream* stream, bool forceCloseAfterSent)
+{
+	if(stream == nullptr) {
+		debug_e("TcpClient::send() ERROR - supplied stream is null");
+		return false;
+	}
+
+	if(this->stream != nullptr) {
+		debug_w("TcpClient::send() - existing stream being discarded");
+		freeStreams();
+	}
+
+	this->stream = stream;
+
+	asyncTotalLen += stream->available();
+	asyncCloseAfterSent = forceCloseAfterSent;
+
+	return true;
+}
+
 bool TcpClient::send(const char* data, uint16_t len, bool forceCloseAfterSent /* = false*/)
 {
 	if(state != eTCS_Connecting && state != eTCS_Connected)
 		return false;
 
-	if(stream == NULL)
-		stream = new MemoryDataStream();
+	if(buffer != nullptr) {
```

Comment:
Should this follow original more closely?

Reply:
> Should this follow original more closely?
> There are a few places I could reduce changes to existing code

I would say yes.